# HG changeset patch
# Parent 661d6c13dcc729ce92bd8134569f29acf1f702c5
We have to make sure that the invisible root node of wxDataViewCtrl is not collapsed.

diff -r 661d6c13dcc7 -r d7d989388f80 src/generic/datavgen.cpp
--- a/src/generic/datavgen.cpp	Mon Oct 29 18:44:05 2012 +0100
+++ b/src/generic/datavgen.cpp	Fri Nov 02 09:48:36 2012 +0100
@@ -347,7 +347,7 @@
     static wxDataViewTreeNode* CreateRootNode()
     {
         wxDataViewTreeNode *n = new wxDataViewTreeNode(NULL, wxDataViewItem());
-        n->SetHasChildren(true);
+        n->m_branchData = new BranchNodeData;
         n->m_branchData->open = true;
         return n;
     }
@@ -416,6 +416,10 @@
 
     void ToggleOpen()
     {
+        // We do not allow the (invisible) root node to be collapsed because there is no way to expand it again.
+        if (m_parent == NULL)
+            return;
+
         wxCHECK_RET( m_branchData != NULL, "can't open leaf node" );
 
         int sum = 0;
@@ -446,6 +450,10 @@
 
     void SetHasChildren(bool has)
     {
+        // We make sure that the (invisible) root node cannot be set to have no children.
+        if (m_parent == NULL)
+            return;
+
         if ( !has )
         {
             wxDELETE(m_branchData);
